src/Microsoft.TemplateEngine.Cli/Commands/TemplateCommand.cs

```diff
@@ -210,20 +216,31 @@ private void AddTemplateOptionsToCommand(CliTemplateInfo templateInfo)
                     {
                         Arity = new ArgumentArity(0, 1)
                     },
-                    ParameterType.Integer => new Option<int>(aliases.ToArray())
+                    ParameterType.Integer => new Option<long>(
+                        aliases.ToArray(),
+                        parseArgument: result => GetParseArgument(parameter, ConvertValueToInt)(result))
                     {
-                        Arity = new ArgumentArity(1, 1)
+                        Arity = new ArgumentArity(0, 1)
```

Comment:
@jonsequitur do we need to set `Arity` in case of custom parse method?

Reply:
You don't need to. It'll be inferred from the type. The arity will determine how many tokens are passed to your parse delegate.